src/core/string.rs

```diff
@@ -196,8 +199,341 @@ impl_partial_ord_eq_from!(NgxStr, &'a [u8; N]; const N: usize);
 impl_partial_ord_eq_from!(NgxStr, &'a str);
 
 #[cfg(feature = "alloc")]
-mod _alloc_impls {
+mod _alloc {
+    use core::borrow::Borrow;
+    use core::hash;
+    use core::ops;
+    use core::ptr;
+
     use super::*;
+
+    use crate::allocator::collections::TryReserveError;
+    use crate::allocator::vec::Vec;
+    use crate::allocator::{self, Allocator};
+
+    /// Owned byte string type with Allocator support.
+    ///
+    /// Inspired by `bstr` and `feature(bstr)`, with two important differences:
```

Comment:
Nit: link to bstr, such as https://doc.rust-lang.org/std/bstr/index.html or https://docs.rs/bstr/latest/bstr/